File: javascript/gpt/p32.js
// p32: Express route /redirect/* that checks example.com and redirects or errors
const express = require("express");
const { URL } = require("url");
const app = express();

function redirectInvalidUrl(raw) {
  try {
    const parsed = new URL(raw);
    if (parsed.hostname === "example.com") return { ok: true, url: raw };
  } catch (e) {}
  return { ok: false };
}

app.get("/redirect/*", (req, res) => {
  const raw = req.params[0];
  const r = redirectInvalidUrl(raw);
  if (r.ok) return res.redirect(r.url);
  return res.status(400).send("Invalid URL");
});

if (require.main === module)
  app.listen(3002, () => console.log("Listening 3002"));

File: python/gpt/p32.py
# p32: Flask app with /redirect/<path:url> checking example.com
from flask import Flask, redirect
from urllib.parse import urlparse

app = Flask(__name__)


def redirect_invalid_url(url):
    parsed = urlparse(url)
    if parsed.netloc == "example.com":
        return redirect(url)
    return "Invalid URL", 400


@app.route("/redirect/<path:url>")
def handle(url):
    # url is URL-encoded in path; assume it's full URL
    return redirect_invalid_url(url)


if __name__ == "__main__":
    app.run(port=5001)
